aws-predictions-tensorflow/src/main/java/com/amplifyframework/predictions/tensorflow/asset/TextClassificationModel.java

```diff
@@ -37,7 +37,7 @@
  * Loads the pre-trained text classification model into
  * a TensorFlow Lite interpreter instance.
  */
-public final class TextClassificationModel implements Loadable<Interpreter, PredictionsException> {
+public class TextClassificationModel implements Loadable<Interpreter, PredictionsException> {
```

Comment:
It might make sense to have a 
```
public interface TextClassificationModel<T> extends Loadable<T, PredictionsException> {
    ...
}
```
with a:
```
public final class TensorFlowTextInterpreter implements TextClassificationModel<Interpreter> {
    ...
}
```

Then, in your test you can `mock(TestClassificationModel.class)`, but in your source, you can keep `TensorflowTextInperpreter`.

Reply:
interfaces are _almost_ always great, but I think we shouldn't overuse it either. `TextClassificationModel` would be a bit _too_ specific of a concept to be appropriate as an interface, and we are not really getting concrete advantage out of it besides allowing its instance to be `final`.

Reply:
Yea, actually, agreed. There aren't going be multiple text classification services -- it's the TensorFlow plugin. The only alternate would be for test purposes. So I do agree, that's overkill.